### DISCUSSION TITLE
GitHub App webhook & autodeploy

### DISCUSSION THREAD

**Discussion by slowlydev** [Dokploy/dokploy#186] [Q&A] [answered]
Hey there, first, thanks for this project, I like it a lot.
There are a few things missing that prevent me from using it.

One thing is the auto deploy. It looks like I have to add the webhook for every app to every repo. Is this correct?
I thought dokploy would handle that via the GitHub App. But it does not look like that, rather it seems as every webhook event gets sent to a site called "webhook.site":

https://github.com/Dokploy/dokploy/blob/faf24dfa25093fbfbb203b5b3d03415c144a81b9/components/dashboard/settings/github/github-setup.tsx#L56-L60

And I am wondering why not the commented out code `${origin}/api/webhook` is getting used but the webhook.site, and If I am not mistaken this is on the main branch as well which gets shipped.

This is a bit confusing to me, but I'd love to implement fully automatic hooks, so I am asking what's holding us back at the moment?

**Comment by Siumauricio** [comment]
Hello, Thanks!

That code in there is purely for testing, in the end I didn't end up using that part, I just forgot to remove it.

Initially it was going to be so automatically, when you install the github app, that it would have all that automated, but in the end I thought that we could have multiple git providers, like bitbucket, gitlab, gitea among others, and I preferred to handle it the other way which is directly here https://github.com/Dokploy/dokploy/blob/canary/pages/api/deploy/%5BrefreshToken%5D.ts#L202, I know it wouldn't take a lot of work to make everything automatic, however the solution we provide is pretty easy to implement and doesn't even take you 3 minutes to implement. https://docs.dokploy.com/en/docs/core/application/auto-deploy#webhook-url

But really there wouldn't be much to do there, it would just be to test and make sure it works and replace that URL with the webhook we have.

**Reply by slowlydev** [reply]
Thanks for the quick reply, okay I see.
Could I try too implement it so that its fully automatic for GitHub? The existing webhook implementation would still work for other git providers other than GitHub right?

**Reply by Siumauricio** [reply, marked as answer]
@slowlydev Not really, it is a little different, because each application has its token with which we identify which application we are going to deploy, however using the webhook that gives github when installing the application sends information on which repository commit is done, I have to validate in each application and docker compose apps, which repository, branch are using and match with the body that sends github, it was not so simple, but well it did not take me long, in the next version should arrive. 

https://github.com/Dokploy/dokploy/pull/188

**Reply by slowlydev** [reply]
Oh wow I didn't expect you to implement it right away thanks a lot!
